subsystem_controllers/src/localization_controller/localization_controller.cpp

```diff
@@ -19,9 +19,158 @@
+    if (sensor_status_.find(alert.source_node) == sensor_status_.end()) {
+      RCLCPP_INFO_STREAM(get_logger(), "Alert not from a sensor node. Alerting node: " << alert.source_node);
+      return;
+    }
+
+    // Update our tracking of sensor statuses if this node failed
+    if (alert.type == carma_msgs::msg::SystemAlert::FATAL) {
+      sensor_status_[alert.source_node] = SensorBooleanStatus::FAILED;
+    }
+
+    // Combine all sensor statuses into a single set to index into the alert map
+    std::vector<SensorBooleanStatus> current_sensor_status;
+    for (const auto& node : config_.sensor_nodes) {
+      current_sensor_status.emplace_back(sensor_status_[node]);
+    }
+
+    // If the controller has not been configured to handle this fault configuration then do nothing
+    if (config_.sensor_fault_map.find(current_sensor_status) == config_.sensor_fault_map.end()) {
+      RCLCPP_INFO_STREAM(get_logger(), "Failure does not match with fault configuration. Assuming system remains operational ");
+      return;
+    }
+
+    // Extract the alert required by this type of sensor statuses
+    SensorAlertStatus alert_status = config_.sensor_fault_map[current_sensor_status];
+    output_alert.description = "Localization required sensor failure from: " + alert.source_node;
+
+    switch(alert_status) {
+      case SensorAlertStatus::FATAL:
+        output_alert.type = carma_msgs::msg::SystemAlert::FATAL;
+        output_alert.description = "Localization subsystem failed due to sensor failure from: " + alert.source_node;
+        break;
+      case SensorAlertStatus::CAUTION:
+        output_alert.type = carma_msgs::msg::SystemAlert::CAUTION;
+        break;
+      case SensorAlertStatus::WARNING:
+        output_alert.type = carma_msgs::msg::SystemAlert::WARNING;
+        break;
+      default:
+      return; // OPERATIONAL case no-op as other subsystems do not need notification of operational status
+    }
+
+    // Publish the alert
+    output_alert.source_node =  get_node_base_interface()->get_fully_qualified_name();
+    publish_system_alert(output_alert);
+  }
+
+  std::unordered_map<std::vector<SensorBooleanStatus>, SensorAlertStatus, VectorHash> LocalizationControllerNode::sensor_fault_map_from_json(std::string json_string) {
+      
+    std::unordered_map<std::vector<SensorBooleanStatus>, SensorAlertStatus, VectorHash> map;
+    
+    rapidjson::Document d;
+
+    if(d.Parse(json_string.c_str()).HasParseError())
+    {
+        RCLCPP_WARN(get_logger(), "Failed to parse sensor_fault map. Invalid json structure");
+        return map;
+    }
+
+    if (!d.HasMember("sensor_fault_map")) {
+      RCLCPP_WARN(get_logger(), "No sensor_fault_map found in localization config");
+      return map;
+    }
+
+    rapidjson::Value& map_value = d["sensor_fault_map"];
+
+    if (!map_value.IsArray()) {
+      RCLCPP_WARN(get_logger(), "No sensor_fault_map cannot be read as array");
```

Comment:
Update to just `"sensor_fault_map cannot be read as array"`

Reply:
fixed